minHeap/minHeap.go

```diff
@@ -0,0 +1,95 @@
+package heap
+
+import (
+	"errors"
+	"fmt"
+)
+
+type heap struct {
+	store    []int
+	capacity int
+	size     int
+}
+
+//creates a heap from an int array
+func buildMinHeap(values []int) (*heap, error) {
+	if len(values) == 0 {
+		return nil, errors.New("Array is empty")
+	}
+	minHeap := heap{capacity: (len(values) * 2), size: 0}
+	h := &minHeap
+	for _, value := range values {
+		h.insert(value)
+		fmt.Println(h.store)
+	}
+	return h, nil
+}
+
+// checks if the array is not full
+func (h *heap) checkCapacity() {
+	if h.capacity <= h.size {
+		h.capacity = h.capacity * 2
+		newStore := make([]int, h.capacity)
+		for _, value := range h.store {
+			newStore = append(newStore, value)
```

Comment:
try to keep references to old store members , to destroy it once its copied.
to decrease the risk of memory leaks.

Reply:
I think golang has automatic garbage collection and once h.store is assigned to the new array, the old one is automatically deleted, am I right?

Reply:
it has a good gc, it will need some tweaking on compiling.